### PR TITLE
Avoid allocating strings during search

### PR DESCRIPTION
**Pull request by jwilder**

### CODE REVIEW

**Comment by otoolep** [comment]
+1

File: tsdb/engine/tsm1/data_file.go [hunk 1]
```diff
@@ -631,10 +630,11 @@ func (d *indirectIndex) Delete(key string) {
 	d.mu.Lock()
 	defer d.mu.Unlock()
 
+	kb := []byte(key)
 	var offsets []int32
 	for _, offset := range d.offsets {
 		_, indexKey, _ := readKey(d.b[offset:])
-		if key == indexKey {
+		if bytes.Compare(kb, indexKey) == 0 {
```
Comment (mark-rushakoff) [inline tsdb/engine/tsm1/data_file.go]: Nitpicky, but I think `bytes.Equal` is more idiomatic that `bytes.Compare(...) == 0`.

Comment (jwilder) [inline tsdb/engine/tsm1/data_file.go]: Good point.  Will update.

**Comment by jwilder** [comment]
Cherry-picked to 0.9.6

**Pull request merged**